src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -302,6 +303,7 @@ private static synchronized ScheduledThreadPoolExecutor threadPool() {
             ws = context.get(FilePath.class);
             node = FilePathUtils.getNodeName(ws);
             DurableTask durableTask = step.task();
+            durableTask.setRun(context.get(Run.class));
```

Comment:
@daniel-beck It looks like a few lines down in this method, you would have access to the `launcher`, so you could call `launcher.prepareFilterRules(context.get(Run.class), durableTask);` (not sure what context makes the most sense, could pass `durableTask` or `this` I guess...) without needing to make any changes in durable-task (other than the addition of `implements EnvVarsFilterableBuilder` on `FileMonitoringTask`, but I am not totally clear on whether it needs to be there or could be here on `DurableTaskStep`).

Any particular reason you wanted to make changes to each kind of task over in `durable-task`? I only glanced at the Jenkins core PR so may very well be missing something obvious.

Reply:
> other than the addition of `implements EnvVarsFilterableBuilder` on `FileMonitoringTask`, but I am not totally clear on whether it needs to be there or could be here on `DurableTaskStep`

I want to allow global rules to determine what they do based on the kind of build step; so I cannot use the same generic type from this plugin for all of {Batch; Shell; Powershell}.